Bottom: e7b95a1f10b817e091f5fcbb8e22b74c6b249bc2
Top:    cf8dfbfe86ca9ba7a86922892ec5b5ff73dfeda4
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-11-18 23:30:43 +0000

Refresh of xi2-hack-test

---

diff --git a/src/hid/gtk/gui-top-window.c b/src/hid/gtk/gui-top-window.c
index 35e2f50..6c12d30 100644
--- a/src/hid/gtk/gui-top-window.c
+++ b/src/hid/gtk/gui-top-window.c
@@ -1711,13 +1711,18 @@ int xi_opcode = -1;
 static void
 handle_motion_event (XIDeviceEvent *event)
 {
-  printf ("XI2 motion event!!\n");
+  double *val;
+  int i;
 
-  if (event == NULL) return;
+  printf ("XI2 motion event!!\n");
 
   printf("    root: %.2f/%.2f\n", event->root_x, event->root_y);
   printf("    event: %.2f/%.2f\n", event->event_x, event->event_y);
 
+  val = event->valuators.values;
+  for (i = 0; i < event->valuators.mask_len * 8; i++)
+      if (XIMaskIsSet(event->valuators.mask, i))
+          printf("        %i: %.2f\n", i, *val++);
 
 }
